src/lib/px4_eigen.h

```diff
@@ -34,18 +34,117 @@
 /**
  * @file px4_eigen.h
  *
- * Compatability header to make Eigen compile on the PX4 stack
+ * @brief Compatibility header to make Eigen compile on the PX4 stack
  * @author Johan Jansen <jnsn.johan@gmail.com>
- */
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ **/
 
 #pragma once
 
+#include <cmath>
+#include <stdio.h>
+#include <mathlib/mathlib.h>
 #pragma GCC diagnostic push
+#ifndef RAND_MAX
 #define RAND_MAX __RAND_MAX
+#endif
 #pragma GCC diagnostic ignored "-Wshadow"
 #pragma GCC diagnostic ignored "-Wfloat-equal"
+#ifndef _GLIBCXX_USE_C99_FP_MACROS_DYNAMIC
 #define _GLIBCXX_USE_C99_FP_MACROS_DYNAMIC 1
+#endif
+
+#include <eigen/Eigen/Eigen>
+// #include <eigen/unsupported/Eigen/CXX11/Tensor>
 
-#include <eigen/Eigen/Core>
-#include <eigen/Eigen/Geometry>
 #pragma GCC diagnostic pop
+
+/**
+ * @brief
+ *	Prints an Eigen::Matrix to stdout
+ **/
+template<typename T>
+static void printEigen(const Eigen::MatrixBase<T> &b)
+{
+	for (int i = 0; i < b.rows(); ++i) {
+		printf("(");
+
+		for (int j = 0; j < b.cols(); ++j) {
+			if (j > 0) {
+				printf(",");
+			}
+
+			printf("%.3f", static_cast<double>(b(i, j)));
+		}
+
+		printf(")%s\n", i + 1 < b.rows() ? "," : "");
+	}
+}
+
+/*
+ * Custom Eigen methods
+ */
+
+/**
+ * @brief
+ *	Construct new Eigen::Quaternion from euler angles
+ *	Right order is YPR.
+ */
+static Eigen::Quaternionf quatFromEuler(const Eigen::Vector3f &rpy){
+	return Eigen::Quaternionf(
+		Eigen::AngleAxisf(rpy.z(), Eigen::Vector3f::UnitZ()) *
+		Eigen::AngleAxisf(rpy.y(), Eigen::Vector3f::UnitY()) *
+		Eigen::AngleAxisf(rpy.x(), Eigen::Vector3f::UnitX())
+		);
+}
+
+/**
+ * @brief
+ *	Construct new Eigen::Vector3f of euler angles from quaternion
+ *	Right order is YPR.
+ */
+static Eigen::Vector3f eulerFromQuat(const Eigen::Quaternionf &q){
+	return q.toRotationMatrix().eulerAngles(2, 1, 0).reverse();
+}
+
+
+ /**
+  * @brief
+  *	Construct new Eigen::Vector3f of euler angles from rotation matrix
+  *	Right order is YPR.
+  */
+static Eigen::Vector3f eulerFromRot(const Eigen::Matrix3f &rot){
+	 return rot.eulerAngles(2, 1, 0).reverse();
+}
+
+/**
+ * @brief
+ *	Construct new Eigen::Matrix3f from euler angles
+ */
+static Eigen::Matrix3f matrixFromEuler(const Eigen::Vector3f &rpy){
+	return quatFromEuler(rpy).toRotationMatrix();
+}
+
+/**
+ * @brief
+ *	Adjust PX4 math::quaternion to Eigen::Quaternionf
+ */
+static Eigen::Quaternionf eigenqFromPx4q(const math::Quaternion &q){
+	return Eigen::Quaternionf(q.data[1], q.data[2], q.data[3], q.data[0]);
```

Comment:
I mean in internal data array. Also perhaps better to use math::Quaternion accessors.